src/commandargs.h

```diff
@@ -19,4 +19,8 @@ typedef std::vector<Setting> ProgramConfiguration;
 
 void ParseCommandArgs(int argc, char* argv[], ProgramConfiguration config);
 
+void Usage(std::vector<SettingOption> options);
```

Comment:
I like this idea. If a Settings struct also contains the array of possible values/argument types then we can create an abstract Usage method which dynamically generates the usage string.

Reply:
Agree! 

This would be the most desirable outcome.